daemons/pacemakerd/pcmkd_messages.c

```diff
@@ -180,7 +180,7 @@ pcmk_ipc_dispatch(qb_ipcs_connection_t * qbc, void *data, size_t size)
 
     } else {
         crm_debug("Unrecognized IPC command '%s' from client %s",
-                  crm_str(task), pcmk__client_name(c));
+                  pcmk__s(task, "(unspecified)"), pcmk__client_name(c));
         pcmk__ipc_send_ack(c, id, flags, "ack", CRM_EX_INVALID_PARAM);
```

Comment:
maybe again pointed ...

Reply:
I think in if-else chains like this we should just catch NULL first so we don't waste time on string comparisons.

BTW the new pcmk__server_command_t design (hash table of IPC command handlers) avoids this entirely ...

Reply:
Good to know! Then there is probably not much sense in optimizing the way of riding dead horses here ;-)